### PR TITLE
Don't strip options from filter when updating relative date values

### LINKED ISSUE #10249: Date range picker unchecks "include today" when date range is changed
**Describe the bug**
Date range picker unchecks "include today" whenever user changes the date range. User has to save twice in order to change a date range AND have the date range include today.

**Logs**

**To Reproduce**
1. Open a report dashboard
2. Open date range data filter 
3. See that "include today" is checked on default 
4. Change the number of days 
5. Click save 

Observe "include today" is unchecked.

**Expected behavior**
"Include today" checkbox remains checked upon save

**Screenshots**
image
"include today" checked.

Change date to 10 days ago, click save.
image
"include today" unchecked.

**Information about your Metabase Installation:**
- Your browser and the version: chrome Version 74.0.3729.169 (Official Build) (64-bit)

- Your operating system: (MacOS)

- Metabase version: version v0.32.4

**Severity**
Annoying for users, must save twice to change date range.

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #10249 

I fixed it just by adding the missing arg. Should I instead use [`updateFilter`](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/query/filter.js#L39)?

### CODE REVIEW

File: frontend/src/metabase/meta/types/Query.js [hunk 1]
```diff
@@ -182,20 +182,13 @@ export type InsideFilter = [
...
-      RelativeDatetimeUnit,
-      FilterOptions,
-    ];
+export type TimeIntervalFilter = [
+  "time-interval",
+  ConcreteField,
+  RelativeDatetimePeriod,
+  RelativeDatetimeUnit,
+  ?TimeIntervalFilterOptions,
+];
```
Comment (paulrosenzweig) [inline frontend/src/metabase/meta/types/Query.js]: I'm not sure why the previous union type didn't work. Flow claimed that it couldn't pick a case, but the arity is different 🤷‍♂ 

Flow seems happier when I consolidated but made the options optional. Does that look ok?

**Review by kdoh — APPROVED** [review]

File: frontend/src/metabase/meta/types/Query.js (hunk 1 above)
Comment (tlrobinson) [reply frontend/src/metabase/meta/types/Query.js]: I actually don't think that's quite correct. It's saying you can have `null` in the last position, but leaving it out entirely will cause an error.

I'll see if I can figure out a solution.

File: frontend/src/metabase/query_builder/components/filters/pickers/RelativeDatePicker.jsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ export default class RelativeDatePicker extends Component {
 
   render() {
     const {
-      filter: [op, field, intervals, unit],
+      filter: [op, field, intervals, unit, options],
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/filters/pickers/RelativeDatePicker.jsx]: The problem with the flow type above is you're destructuring `options` even if it's the 4 arity form of `TimeIntervalFilter`.

File: frontend/src/metabase/query_builder/components/filters/pickers/RelativeDatePicker.jsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ export default class RelativeDatePicker extends Component {
             typeof intervals === "number" ? Math.abs(intervals) : intervals
           }
           onChange={value =>
-            onFilterChange([op, field, formatter(value), unit])
+            onFilterChange([op, field, formatter(value), unit, options])
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/filters/pickers/RelativeDatePicker.jsx]: This is going to add a `null` to the end of the filter even if it doesn't have options.

Ideally we'd expand the `Filter` wrapper class in `metabase-lib` to have various subclasses like we do for `Dimension` (which should probably be called `FieldReference`, FWIW) so you could just do `filter.setPeriod()` or `filter.setUnit()` etc which would carry over options if present. This kind of logic shouldn't be in React components to begin with.

I went ahead and pushed a kind of hacky solution just using `assoc` for now.

**Review by tlrobinson — APPROVED** [review]

**Pull request merged**